### PR TITLE
Add structure gtfs realtime in Vehicle_occupancies

### PR DESCRIPTION
**Pull request by julyto**
Creation of the structure to integrate a GTFS-RT flow in the Vehicle_occupancies API.

- declaration of the vehicleOccupancies interface
- creation of the factory
- declaration of structs and methods (no implemented)

### CODE REVIEW

File: internal/vehicleoccupancies/factory.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+// and to create them according to their base type.
+// To implement an interface, the contexts must imperatively implement all the methods declared in it.
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+)
+
+type IVehicleOccupancy interface {
```
Comment (benoit-bst) [inline internal/vehicleoccupancies/factory.go:13]: :+1:

File: internal/vehicleoccupancies/factory.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	RefreshVehicleOccupanciesLoop(predictionURI url.URL,
+		externalToken string, navitiaURI url.URL, navitiaToken string, loadExternalRefresh,
+		connectionTimeout time.Duration, location *time.Location)
+
+	LoadDataExternalSource(uri url.URL, token string,
+		connectionTimeout time.Duration, location *time.Location) (*GtfsRt, error)
+}
+
+// Patern factory Vehicle occupancies
+func VehicleOccupancyFactory(type_vehicleoccupancy string) (IVehicleOccupancy, error) {
```
Comment (benoit-bst) [inline internal/vehicleoccupancies/factory.go:23]: I don't understand this part. You return a `IVehicleOccupancy` but when i see https://github.com/CanalTP/forseti/pull/34/files#diff-ba7345e8d844a1f2fe007ef4ec62d9e2fc6c0d610c1a366494f27020514d99d2R25, this is not an interface but a context structure. 
Does it work ?

Comment (julyto) [reply internal/vehicleoccupancies/factory.go:23]: Yes it work, in Go the notion of Interface just ask to implement the methods declared. For us the context use the two methods, it's considerate as Interface and will be used in main() as well.

**Review by benoit-bst — APPROVED** [review]

**Review by kadhikari — APPROVED** [review]
Great work

**Pull request merged**